lib/fake_ifdh.py

```diff
@@ -75,10 +79,14 @@ def getRole(role_override: Optional[str] = None, verbose: int = 0) -> str:
             return role
 
     # if there's a role in the wlcg.groups of the token, pick that
-    if os.environ.get("BEARER_TOKEN_FILE", False):
-        with os.popen("decode_token.sh $BEARER_TOKEN_FILE", "r") as f:
-            token_s = f.read()
-            token = json.loads(token_s)
+    if os.environ.get("BEARER_TOKEN_FILE", False) and os.path.exists(
+        os.environ["BEARER_TOKEN_FILE"]
+    ):
+        with open(os.environ["BEARER_TOKEN_FILE"]) as f:
+            token_encoded = f.read(4096).strip()
+            token = dict(
```

Comment:
Why do you convert the claims to a dict? Why not just call the `get()` method of the scitoken object itself? https://scitokens.readthedocs.io/en/latest/scitokens.html#scitokens.scitokens.SciToken.get

Reply:
The dict bit is to help make mypy happy.   There isn't a typing stub for scitokens (or at least I didn't find one) so
that makes it a Dict[Any,Any] and then we convert the Any  we extract with str() when we fetch it and mypy is happy.
I went about 4 rounds with other permutations trying to get mypy to not  choke on it.

Also, you can't do x['key'] directly on a x = SciToken(), but you can x.claims()['key'] , but you can't do x.claims().get('key')... 
but you can do dict(x.claims()).get("key") AND dict(x.claims)["key"].   At least that's what I remember
from my interactive tests.  So I call claims() and convert it to a dict everywhere, and I can use either
.get(k) or [k] on it.

Reply:
> Also, you can't do x['key'] directly on a x = SciToken(), but you can x.claims()['key'] , but you can't do x.claims().get('key')...

You _can_ call `x.get('key')` though.

Reply:
This should all be cleaned up now.